lib/routes/routeBackbeat.js

```diff
@@ -482,6 +482,7 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
             // Retrieve the null version id from the object metadata.
             versionId = objMd && objMd.versionId;
             if (!versionId) {
+                omVal.isNull = true;
```

Comment:
Could be worth commenting what this does or is meant to do as it does not look obvious

Reply:
I agree, I'm just still looking for the piece of code that this change affects downstream.

Reply:
Added a comment, but posting it here as well for visibility.
The piece of code that changes behavior is this [here](https://github.com/scality/cloudserver/blob/328b7bc335f10b8e88dd9553884507ad96df77c5/lib/metadata/metadataUtils.js#L20). During a get object for the "null" versionId cloudserver fetches the master version and determines whether it is a null version using the check
```
objMD.isNull || !objMD.versionId
``` 
Without this fix after after the update you would be left with an object that now has an versionId with `isNull` as undefined.
This would prevent cloudserver from recognizing this as a null version and cause it to return a 404.